sources/api/netdog/src/dns.rs

```diff
@@ -141,6 +152,79 @@ impl DnsSettings {
 
         fs::write(path, output).context(error::ResolvConfWriteFailedSnafu { path })
     }
+
+    /// Write a drop-in file for systemd-resolved
+    #[cfg(net_backend = "systemd-networkd")]
+    pub(crate) fn write_resolved_dropin(&self) -> Result<()> {
+        Self::write_resolved_dropin_impl(self, RESOLVED_CONF_DROPIN)
+    }
+
+    #[cfg(net_backend = "systemd-networkd")]
+    fn write_resolved_dropin_impl<P>(&self, path: P) -> Result<()>
+    where
+        P: AsRef<Path>,
+    {
+        let path = path.as_ref();
+        let resolved_dropin = ResolvedConfDropin::from_dns_settings(&self);
+
+        fs::write(path, resolved_dropin.to_string())
+            .context(error::ResolvConfWriteFailedSnafu { path })
+    }
+
+    #[cfg(net_backend = "systemd-networkd")]
+    pub(crate) fn has_name_servers(&self) -> bool {
+        self.nameservers.is_some()
+    }
+
+    #[cfg(net_backend = "systemd-networkd")]
+    pub(crate) fn has_search_domains(&self) -> bool {
+        self.search.is_some()
+    }
+}
+
+#[cfg(net_backend = "systemd-networkd")]
+#[derive(Debug, SystemdUnit)]
+struct ResolvedConfDropin {
+    resolve: Option<ResolveSection>,
+}
+
+#[cfg(net_backend = "systemd-networkd")]
+#[derive(Debug, SystemdUnitSection)]
+#[systemd(section = "Resolve")]
+struct ResolveSection {
+    #[systemd(entry = "DNS", space_separated)]
+    dns: Vec<String>,
+    #[systemd(entry = "Domains", space_separated)]
+    domains: Vec<String>,
+}
+
+#[cfg(net_backend = "systemd-networkd")]
+impl ResolvedConfDropin {
+    fn from_dns_settings(dns: &DnsSettings) -> Self {
+        let domains = if let Some(domains) = &dns.search {
+            domains.clone()
+        } else {
+            Vec::new()
+        };
```

Comment:
Not really in this case.  The derive macro operates on `Vec`s and `Option`s with values, not nested objects like `Option<Vec<_>>`. (Which was purposeful)

I don't love creating an empty Vec, but I also feel that an empty `Vec` has the same significance in this case as `None`, and far easier to handle in code than a `Some` with an empty `Vec` inside of it.

Reply:
Oh - or were you suggesting that the field in `ResolveSection` be an `Option`?  Like:
```
ResolveSection {
    dns: Option<...>
}
```

Even then, I'd still prefer a `Vec` since the field needs to be space separated and `Option<String>` becomes hard to disambiguate.

But perhaps I'm missing your point.